File: Easy/Power Set.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
	public:
		void ric(int i, string &s, vector<string> &ans, string stringa)
		{
		    if (i==s.length())
		    {
		        ans.push_back(stringa);
		        return;
		    }
		    
		    ric(i+1, s, ans, stringa);
		    ric(i+1, s, ans, stringa+s[i]);
		}
		
		
		vector<string> AllPossibleStrings(string s)
		{
		    // Code here
		    vector<string> ans;
		    
		    ric(0,s,ans,"");
		    sort (ans.begin(), ans.end());
		    ans.erase(ans.begin());
		    return ans;
		}
};

//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		string s;
		cin >> s;
		Solution ob;
		vector<string> res = ob.AllPossibleStrings(s);
		for(auto i : res)
			cout << i <<" ";
		cout << "\n";

	}
	return 0;
}
// } Driver Code Ends

File: Medium/Power Set.py
#User function Template for python3
import itertools

class Solution:
	def AllPossibleStrings(self, s):
		# Code here
        n = len(s)
		ans = []
    
        for i in range(1, 2 ** n):
            temp = ""
            
            for j in range(n):
                if (i >> j) & 1:
                    temp += s[j]
            
            ans.append(temp)
        
        return sorted(ans)
        
#{ 
 # Driver Code Starts

#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		s = input()
		ob = Solution();
		ans = ob.AllPossibleStrings(s)
		for i in ans:
			print(i, end = " ");
		print()
# } Driver Code Ends
